stdlib/format.ml

```diff
@@ -391,6 +415,32 @@ let format_pp_token state size = function
       end
     end
 
+  | Pp_custom_break { no_break; yes_break } ->
+    begin match Stack.top_opt state.pp_format_stack with
+    | None -> () (* No open box. *)
+    | Some { box_type; width } ->
+      begin match box_type with
+      | Pp_hovbox ->
+        if size > state.pp_space_left
+        then custom_break_new_line state yes_break width
+        else custom_break_same_line state no_break
+      | Pp_box ->
+        (* Have the line just been broken here ? *)
+        if state.pp_is_new_line then custom_break_same_line state no_break else
+        if size > state.pp_space_left
+         then custom_break_new_line state yes_break width else
+        (* break the line here leads to new indentation ? *)
+        let off = String.length (snd yes_break) in
+        if state.pp_current_indent > state.pp_margin - width + off
+        then custom_break_new_line state yes_break width
+        else custom_break_same_line state no_break
+      | Pp_hvbox -> custom_break_new_line state yes_break width
+      | Pp_fits -> custom_break_same_line state no_break
+      | Pp_vbox -> custom_break_new_line state yes_break width
+      | Pp_hbox -> custom_break_same_line state no_break
+      end
+    end
```

Comment:
Also here, if the change is approved it would be better to refactor to avoid repetition with the `Pp_break` case. Or just remove or commandeer `Pp_break`.